### DISCUSSION TITLE
What is the equation to convert raw Pet EXP to Pet Level

### DISCUSSION THREAD

**Discussion by HammyInTheWall** [HypixelDev/PublicAPI#525] [Q&A]
I collected the values of exp required to get each pet level, as well as the cumulative pet exp required for each level. I graphed this data and can't seem to find an equation to the graph that fits well enough. Rather than do tons and tons of math to try and figure it out, I was wondering if any devs could let me know the equation or what the conversion is, considering that raw exp is converted to pet level in-game (plus plancke's site does so as well on the leaderboard).

Thanks

**Comment by Picsou993** [comment]
Hi, for lobby pets you can use the array of exp needed to level up that you can find here:
https://github.com/HypixelDev/PublicAPI/blob/d77a9d88f308bc07dada4636b6dca3439dcc8efb/hypixel-api-core/src/main/java/net/hypixel/api/pets/Pet.java#L7-L14

**Reply by HammyInTheWall** [reply]
I already have these values, I was more wondering if there's any form of equation you can use to solve for level on total exp, or if you actually just have to take current exp and subtract each level to solve for current level and % progress to next level

**Reply by HammyInTheWall** [reply]
Looking at the page further I answered my own question, it looks like they literally do just figure out the level and then subtract the remaining experience needed from the requirement of the next level, rip. Oh well. Thanks for the link Picsou.
